command/debug/debug.go

```diff
@@ -0,0 +1,613 @@
+func (c *cmd) configuredTarget(target string) bool {
+	for _, dt := range c.capture {
+		if dt == target {
+			return true
+		}
+	}
+	return false
+}
+
+// createArchive walks the files in the temporary directory
+// and creates a tar file that is gzipped with the contents
+func (c *cmd) createArchive() error {
+	f, err := os.Create(c.output + debugArchiveExtension)
+	if err != nil {
+		return fmt.Errorf("failed to create compressed archive: %s", err)
+	}
+	defer f.Close()
+
+	g := gzip.NewWriter(f)
+	defer g.Close()
+	t := tar.NewWriter(f)
+	defer t.Close()
+
+	err = filepath.Walk(c.output, func(file string, fi os.FileInfo, err error) error {
+		if err != nil {
+			return fmt.Errorf("failed to walk filepath for archive: %s", err)
+		}
+
+		header, err := tar.FileInfoHeader(fi, fi.Name())
+		if err != nil {
+			return fmt.Errorf("failed to create compressed archive header: %s", err)
+		}
+
+		header.Name = filepath.Join(filepath.Base(c.output), strings.TrimPrefix(file, c.output))
+
+		if err := t.WriteHeader(header); err != nil {
+			return fmt.Errorf("failed to write compressed archive header: %s", err)
+		}
+
+		// Only copy files
+		if !fi.Mode().IsRegular() {
+			return nil
+		}
+
+		f, err := os.Open(file)
+		if err != nil {
+			return fmt.Errorf("failed to open target files for archive: %s", err)
+		}
+
+		if _, err := io.Copy(t, f); err != nil {
+			return fmt.Errorf("failed to copy files for archive: %s", err)
+		}
+
+		f.Close()
+
+		return nil
+	})
+
+	if err != nil {
+		return fmt.Errorf("failed to walk output path for archive: %s", err)
+	}
+
+	// Remove directory that has been archived
+	err = os.RemoveAll(c.output)
+	if err != nil {
+		return fmt.Errorf("failed to remove archived directory: %s", err)
+	}
+
+	return nil
+}
+
+// defaultTargets specifies the list of all targets that
+// will be captured by default
+func (c *cmd) defaultTargets() []string {
+	return append(c.dynamicTargets(), c.staticTargets()...)
+}
+
+// dynamicTargets returns all the supported targets
+// that are retrieved at the interval specified
+func (c *cmd) dynamicTargets() []string {
+	return []string{"metrics", "logs", "pprof"}
+}
+
+// staticTargets returns all the supported targets
+// that are retrieved at the start of the command execution
+func (c *cmd) staticTargets() []string {
+	return []string{"host", "agent", "cluster"}
+}
+
+func (c *cmd) Synopsis() string {
+	return synopsis
+}
+
+func (c *cmd) Help() string {
+	return c.help
+}
+
+const synopsis = "Monitors a Consul agent for the specified period of time, recording information about the agent, cluster, and environment to an archive written to the relative directory."
+const help = `
+Usage: consul debug [options]
+
+  Monitors a Consul agent for the specified period of time, recording
+  information about the agent, cluster, and environment to an archive
+  written to the relative directory.
+
+  If ACLs are enabled, an agent token must be supplied in order to perform
+  this operation.
```

Comment:
Suggest we make it a token with `operator:read`. It's not clear in any case what an "agent token" is as there are a few different token configs related to agents.